runtime-testsuite/test/org/antlr/v4/test/runtime/FileUtils.java

```diff
@@ -55,7 +61,7 @@ public static void mkdir(String dir) {
 	}
 
 	public static void deleteDirectory(File f) throws IOException {
-		if (f.isDirectory()) {
+		if (f.isDirectory() && !isLink(f.toPath())) {
```

Comment:
Could you clarify why link checking during deleting is important? What does it affect on?

Reply:
During tests setup, the Antlr package is webpacked.
For each test, that package is installed as a dependency of the project in node_modules, using npm link.
Without this change, the antlr package gets deleted as part of destroying the test folder, which makes all subsequent tests fail.